rpc/src/v1/helpers/dispatch.rs

```diff
@@ -449,28 +480,60 @@ fn sign_transaction(
 #[derive(Debug, Clone, Copy)]
 enum ProspectiveSignerState {
 	TryProspectiveSign,
+	WaitForPostSign,
 	WaitForNonce,
-	Finish,
 }
 
-struct ProspectiveSigner {
+struct ProspectiveSigner<P: PostSign> {
 	accounts: Arc<AccountProvider>,
 	filled: FilledTransactionRequest,
 	chain_id: Option<u64>,
 	reserved: nonce::Reserved,
 	password: SignWith,
 	state: ProspectiveSignerState,
-	prospective: Option<Result<WithToken<SignedTransaction>>>,
+	prospective: Option<WithToken<SignedTransaction>>,
 	ready: Option<nonce::Ready>,
+	post_sign: Option<P>,
+	post_sign_future: Option<<P::Out as IntoFuture>::Future>
 }
 
-impl ProspectiveSigner {
+/// action to execute after signing
+/// e.g importing a transaction into the chain
+pub trait PostSign: Send {
```

Comment:
Just a quick idea (maybe dumb idea :) , maybe this trait could be more generic. PostSign is specific name, but as I understand it is a definition for a future post action on a signed transaction, if we make the signed transaction (`WithToken<SignedTransaction>`) an inner type (input) we got  something that is totally generic (I wonder if it maybe already exists in the future api something like a trait for implementing a combination of `future::ok` and `and_then`).

Reply:
the futures crate has no such api